lib/ukcpuid/include/uk/cpuid.h

```diff
@@ -0,0 +1,68 @@
+#ifndef __UK_CPUID_H__
+#define __UK_CPUID_H__
+
+#include <uk/list.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define cpuid_macro(ax, bx, cx, dx, func_1, func_2) do { \
```

Comment:
I would split internal definitions into an internal header and only list public accessible definitions and interfaces in this header file.

Reply:
Sounds like a good idea :like